students/eowyn/session06/mailroom/test_mailroom.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import mailroom
+import pytest
+import os.path
+
+
+def test_safeinput():
+    result = mailroom.safe_input()
+    assert result is None
+
+
+def test_return_to_menu():
+    result = mailroom.return_to_menu()
+    assert result is True
+
+
+def test_remove_inputquotes():
+    astring = "'hello'"
+    output = 'hello'
+    assert mailroom.remove_inputquotes(astring) == output
+
+
+def test_generate_letter():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    output = 'Thank you, Margaret Atwood, for your generosity and recent gift of $555.'
```

Comment:
be a bit careful about testing for exact results.

If you change the letter template a bit, it will fail --not too bad, you can update the test. Bu tyou may want to test if critical parts of the letter are there -- the right name, the right dollar amount, rather than all the text.